app/kingdom/contest.py

```diff
@@ -0,0 +1,53 @@
+from datetime import date
+
+from app.people.knight import Knight
+
+
+class Contest:
+    def __init__(self, date: date, king: str, knights: dict) -> None:
+        self.date = date
+        self.king = king
+        self.knights = knights
+        print(f"On the {date} Fight contest of {len(knights)} "
+              f"knights was created by King {king}")
+
+    def prepare_for_battle(self) -> None:
+        for fighter in self.knights.values():
+            fighter.protection = 0
+            for arm in fighter.armour:
+                fighter.protection += arm["protection"]
+            fighter.power += fighter.weapon["power"]
+            if fighter.potion is not None:
+                if "power" in fighter.potion["effect"]:
+                    fighter.power += fighter.potion["effect"]["power"]
+
+                if "protection" in fighter.potion["effect"]:
+                    fighter.protection +=\
```

Comment:
The line continuation character `\` is used here. It's generally better to use parentheses for line continuation to improve readability and avoid potential issues.